components/kyma-environment-broker/internal/runtime/handler.go

```diff
@@ -148,6 +123,157 @@ func (h *Handler) takeLastNonDryRunClusterOperations(oprs []internal.UpgradeClus
 	return toReturn, totalCount
 }
 
+func setRuntimeStateByOperationState(dto *pkg.RuntimeDTO) {
+	// Determine runtime state based on the last operation of the runtime
+	lastOp := dto.LastOperation()
+	switch lastOp.State {
+	case string(domain.Succeeded):
+		dto.Status.State = pkg.StateSucceeded
+		if lastOp.Type == pkg.Suspension {
+			dto.Status.State = pkg.StateSuspended
+		}
+	case string(domain.Failed):
+		dto.Status.State = pkg.StateFailed
+		if lastOp.Type == pkg.UpgradeKyma || lastOp.Type == pkg.UpgradeCluster {
+			dto.Status.State = pkg.StateError
+		}
+	case string(domain.InProgress):
+		switch lastOp.Type {
+		case pkg.Provision, pkg.Unsuspension:
+			dto.Status.State = pkg.StateProvisioning
+		case pkg.Deprovision, pkg.Suspension:
+			dto.Status.State = pkg.StateDeprovisioning
+		case pkg.UpgradeKyma, pkg.UpgradeCluster:
+			dto.Status.State = pkg.StateUpgrading
```

Comment:
let's add here an Upgrade type also (the PR should be merged soon, see update process  https://github.com/kyma-project/control-plane/pull/795)